File: stripe/src/main/java/com/stripe/android/view/AddPaymentMethodCardRowView.java
package com.stripe.android.view;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.support.annotation.NonNull;

import com.stripe.android.R;
import com.stripe.android.model.PaymentMethod;

@SuppressLint("ViewConstructor")
public class AddPaymentMethodCardRowView extends AddPaymentMethodRowView {
    AddPaymentMethodCardRowView(@NonNull final Activity activity,
                                @NonNull final PaymentMethodsActivityStarter.Args args) {
        super(activity,
                R.layout.add_payment_method_card_row,
                R.id.payment_methods_add_card,
                new AddPaymentMethodActivityStarter.Args.Builder()
                        .setShouldAttachToCustomer(true)
                        .setShouldRequirePostalCode(args.shouldRequirePostalCode)
                        .setIsPaymentSessionActive(args.isPaymentSessionActive)
                        .setPaymentMethodType(PaymentMethod.Type.Card)
                        .setPaymentConfiguration(args.paymentConfiguration)
                        .build());
    }
}

File: stripe/src/main/java/com/stripe/android/view/AddPaymentMethodCardRowView.kt
package com.stripe.android.view

import android.annotation.SuppressLint
import android.app.Activity
import com.stripe.android.R
import com.stripe.android.model.PaymentMethod

@SuppressLint("ViewConstructor")
internal class AddPaymentMethodCardRowView internal constructor(
    activity: Activity,
    args: PaymentMethodsActivityStarter.Args
) : AddPaymentMethodRowView(
    activity,
    R.layout.add_payment_method_card_row,
    R.id.stripe_payment_methods_add_card,
    AddPaymentMethodActivityStarter.Args.Builder()
        .setBillingAddressFields(args.billingAddressFields)
        .setShouldAttachToCustomer(true)
        .setIsPaymentSessionActive(args.isPaymentSessionActive)
        .setPaymentMethodType(PaymentMethod.Type.Card)
        .setAddPaymentMethodFooter(args.addPaymentMethodFooterLayoutId)
        .setPaymentConfiguration(args.paymentConfiguration)
        .setWindowFlags(args.windowFlags)
        .build()
)
